### DISCUSSION TITLE
Text going on the wrong side of the images.

### DISCUSSION THREAD

**Discussion by JohnPhillips9075** [ImageMagick/ImageMagick#6739] [Digital Image Processing]
I am converting a bunch of images and adding text to the bottom of the images (or that is what I want to do) only on some images it is going on the side rather than the bottom.

I have attached the before and after of each image... why are they going on the ends and not the bottom.   These are select files which cause the problem for the most part I have lots more that work as desired on the bottom and many that end up on the east and west walls of the image.

Thanks for your help.

Here is a script I use to process them all.   

echo off
for %%f in (*.jpg) do call :Convert %%f
Goto :EOF
:Convert
echo %1
"C:\Program Files\ImageMagick-7.1.1-Q16-HDRI\magick" convert "%1" -background Khaki   -gravity center -size 480 -pointsize 50 pango:"Buzz Art: This is a test" -append "%~n1-out.jpg"
Goto :EOF

![Chairs_Names_2021103](https://github.com/ImageMagick/ImageMagick/assets/111818026/b4140147-0340-415f-a778-0000f032378f)
![Chairs_Names_2021103-out](https://github.com/ImageMagick/ImageMagick/assets/111818026/239e34ef-cf8a-40c0-815e-0825abd7e6cc)
![Gym_Logo_2021115](https://github.com/ImageMagick/ImageMagick/assets/111818026/5176c24a-365c-4c3e-946f-1c3c73314fea)
![Gym_Logo_2021115-out](https://github.com/ImageMagick/ImageMagick/assets/111818026/a0411175-e285-46dc-99b1-951cf2a885a2)
![Hair_Dresser_2021116](https://github.com/ImageMagick/ImageMagick/assets/111818026/3e609263-50fd-47d5-ae14-0715037d3b7d)
![Hair_Dresser_2021116-out](https://github.com/ImageMagick/ImageMagick/assets/111818026/d24ce988-5ec5-491a-8b63-5d7df570417a)
![Happy_PLace_2021118](https://github.com/ImageMagick/ImageMagick/assets/111818026/fb60cc66-1e17-4e76-88c2-466323fcb826)
![Happy_PLace_2021118-out](https://github.com/ImageMagick/ImageMagick/assets/111818026/e1fa0598-86be-42c7-9693-e1e215338be5)

**Comment by snibgo** [comment]
Your images have orientation metadata. I suggest inserting "-auto-orient" immediately after reading the input file, `%1`.

I also suggest removing `convert` from your command, so you run `magick`, not `magick convert`.

When you include `convert`, IM will use the old v6 syntax instead of the superior v7 syntax.

**Comment by fmw42** [comment]
JPG image contain EXIF header information about the orientation that the camera was in when the image was captured.  Add -auto-orient to your command right after reading the input. That will correct the rotation of the input image.

Note some viewers may not use this information and the image will be displayed incorrectly when you look at it.

**Comment by JohnPhillips9075** [comment]
Your guys are amazing... works perfectly for every image.   Thanks guys.   Reposting final script for anyone in the future with this issue.  Kinda of figured it had something to do with camera orientation just could not figure out how to fix it easily.

echo off
for %%f in (*.jpg) do call :Convert %%f
Goto :EOF

:Convert
echo %1
"C:\Program Files\ImageMagick-7.1.1-Q16-HDRI\magick" "%1" -auto-orient -background Khaki   -gravity center -size 480 -pointsize 50 pango:"Buzz Art: This is a test" -append "%~n1-out.jpg"
Goto :EOF
